src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -551,6 +551,21 @@ private bool ValidatePackagesSha512(PackagesLockFile lockFile, LockFile assetsFi
             var isLockFileValid = false;
             var success = true;
 
+            var isLockedMode = _request.Project.RestoreMetadata.RestoreLockProperties.RestoreLockedMode;
+            if (isLockedMode && _request.RestoreForceEvaluate)
+            {
+                success = false;
+
+                // invalid input since LockedMode and RestoreForce should not be used together.
+                var message = string.Format(CultureInfo.CurrentCulture, Strings.Error_RestoreLockedModeWithForceEvaluate, packagesLockFilePath);
+
+                // directly log to the request logger when we're not going to rewrite the assets file otherwise this log will
+                // be skipped for netcore projects.
+                await _request.Log.LogAsync(RestoreLogMessage.CreateError(NuGetLogCode.NU1004, message));
```

Comment:
If you look at the log code, it refers to dependency changes, but this is more of an input thing, but this is closer to https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1005. 

I think we either reuse NU1005 or add a new one. 
What do you think?

Reply:
Hmm I think we can reuse NU1005 because is related to `RestorePackagesWithLockFile` but the error is more related to the use of commands so we can go either way.

Reply:
It's conflicting/bad input. 

NU1005 was gonna be my suggestion as well. 

We can document the 2 instances where that warning is raised.